project-base/storefront/components/Pages/Order/TransportAndPayment/TransportAndPaymentSkeletonContext.tsx

```diff
@@ -0,0 +1,71 @@
+import { useCurrentCart } from 'connectors/cart/Cart';
+import { getInternationalizedStaticUrls } from 'helpers/getInternationalizedStaticUrls';
+import { useDomainConfig } from 'hooks/useDomainConfig';
+import { useRouter } from 'next/router';
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import { usePersistStore } from 'store/usePersistStore';
+
+type TransportAndPaymentSkeletonContextType = {
+    isTransportAndPaymentSkeletonVisible: boolean;
+    setIsLoadingInitialTransportAndPaymentData: React.Dispatch<React.SetStateAction<boolean>>;
+    setIsLoadingTransportAndPaymentFromLastOrder: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+const initialContextValue: TransportAndPaymentSkeletonContextType = {
+    isTransportAndPaymentSkeletonVisible: true,
+    // eslint-disable-next-line no-empty-function, @typescript-eslint/no-empty-function
+    setIsLoadingInitialTransportAndPaymentData: () => {},
+    // eslint-disable-next-line no-empty-function, @typescript-eslint/no-empty-function
+    setIsLoadingTransportAndPaymentFromLastOrder: () => {},
+};
+
+const TransportAndPaymentSkeletonContext = createContext<TransportAndPaymentSkeletonContextType>(initialContextValue);
```

Comment:
Why use Context and not hooks with Zustand? I would avoid using Context until there's reason for it. And even though I tried to find reason for it, I couldn't find any. Also Zustand usage is much cleaner then using Context.

Reply:
To be honest I dont have a clear candidate. The reason for me was that this way it is clearly separated from the rest of the application. However, I agree that a global store is much easier to work with and it does not require component restructuring just to make the context available. I rewrote it to Zutand, then. However, let's still discuss it in a FE meeting.